src/hotspot/share/services/management.cpp

```diff
@@ -1705,10 +1700,10 @@ JVM_ENTRY(jint, jmm_GetInternalThreadTimes(JNIEnv *env,
 
   ThreadTimesClosure ttc(names_ah, times_ah);
   {
-    MutexLocker ml(THREAD, Threads_lock);
-    Threads::threads_do(&ttc);
+    JavaThreadIteratorWithHandle jtiwh;
+    Threads::threads_do(&jtiwh, &ttc);
```

Comment:
Same comments as above re need for Threads_lock.

Reply:
I'm backing out the changes to src/hotspot/share/services/management.cpp.